In [1]:
import pandas as pd
from sklearn import linear_model

df = pd.read_csv("cars.csv")

X = df[['Weight', 'Volume']]
X.columns = ['Weight', 'Volume']  # set the column names to match the feature names used in the model fit
y = df['CO2']

regr = linear_model.LinearRegression()
regr.fit(X, y)

#predict the CO2 emission of a car where the weight is 2300kg, and the volume is 1300cm3:
predictedCO2 = regr.predict([[2300, 1300]])

print(predictedCO2)

print(regr.coef_)


[107.2087328]
[0.00755095 0.00780526]


/opt/conda/lib/python3.10/site-packages/sklearn/base.py:439: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(


These values tell us that if the weight increase by 1kg, the CO2 emission increases by 0.00755095g.
And if the engine size (Volume) increases by 1 cm3, the CO2 emission increases by 0.00780526 g.

In [2]:
#predict the CO2 emission of a car where the weight is 3300kg, and the volume is 1300cm3:
predictedCO2 = regr.predict([[3300, 1300]])

print(predictedCO2)

# print(regr.coef_)

[114.75968007]


/opt/conda/lib/python3.10/site-packages/sklearn/base.py:439: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(


We have predicted that a car with 1.3 liter engine, and a weight of 3300 kg, 
will release approximately 115 grams of CO2 for every kilometer it drives.

Validate coefficient
107.2087328 + (1000 * 0.00755095) = 114.75968